From 7d75ef5f56dbc2bedb9a893eb9ecfd03c456338d Mon Sep 17 00:00:00 2001
From: Flavio Silveira <fggs@terra.com.br>
Date: Thu, 29 Aug 2013 08:51:48 -0300
Subject: [PATCH v2] mtd: m25p80: Add support for ESMT F25L32PA

This flashchip is used in D-Link DIR-610 A1 router board 
and maybe several others, yet is not kernel upstream.

So add support for it according to datasheet [0], making it easier
to support other boards using this flashchip in the future.

Changelog v2:
 - Better description
 - Datasheet link at the bottom, similar to other patches.

[0] http://www.esmt.com.tw/DB/manager/upload/F25L32PA.pdf 

Signed-off-by: Flavio Silveira <fggs@terra.com.br>
---
 drivers/mtd/devices/m25p80.c |    3 +++
 1 file changed, 3 insertions(+)

 http://patchwork.ozlabs.org/patch/272438/
 http://lists.infradead.org/pipermail/linux-mtd/2013-September/048511.html

--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -908,6 +908,9 @@ static const struct spi_device_id m25p_i
 	/* ESMT */
 	{ "f25l32pa", INFO(0x8c2016, 0, 64 * 1024, 64, SECT_4K) },
 
+	/* ESMT */
+	{ "f25l32pa", INFO(0x8c2016, 0, 64 * 1024, 64, SECT_4K) },
+
 	/* Everspin */
 	{ "mr25h256", CAT25_INFO( 32 * 1024, 1, 256, 2, M25P_NO_ERASE | M25P_NO_FR) },
 	{ "mr25h10",  CAT25_INFO(128 * 1024, 1, 256, 3, M25P_NO_ERASE | M25P_NO_FR) },
